src/darwin/Framework/CHIP/CHIPOperationalCredentialsDelegate.mm

```diff
@@ -201,9 +201,14 @@ static BOOL isRunningTests(void)
         return CHIP_ERROR_INTERNAL;
     }
 
-    X509CertRequestParams noc_request = { 1, mIssuerId, validityStart, validityEnd, true, fabricId, true, nodeId };
-    ReturnErrorOnFailure(
-        NewNodeOperationalX509Cert(noc_request, CertificateIssuerLevel::kIssuerIsRootCA, pubkey, *mIssuerKey, noc));
+    ChipDN noc_dn;
+    noc_dn.AddAttribute(chip::ASN1::kOID_AttributeType_ChipFabricId, fabricId);
+    noc_dn.AddAttribute(chip::ASN1::kOID_AttributeType_ChipNodeId, nodeId);
```

Comment:
We have several copies of this code, which means we have multiple places (and more to come?) where consumers have to get this right, including the exact IODs to use, etc.

As a followup: Is there any way to make that less error-prone?

Reply:
Added TODO to followup